backend/src/schema/resolvers/searches.js

```diff
@@ -0,0 +1,74 @@
+import log from './helpers/databaseLogger'
+
+export default {
+  Query: {
+    findResources: async (_parent, args, context, _resolveInfo) => {
+      const { query, limit } = args
+      const { id: thisUserId } = context.user
+      // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
+      const myQuery = query
+        .replace(/\s+/g, ' ')
+        .replace(/[[@#:*~\\$|^\]?/"'(){}+?!,.-;]/g, '')
+        .split(' ')
+        .map(s => (s.toLowerCase().match(/^(not|and|or)$/) ? '"' + s + '"' : s + '*'))
```

Comment:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2019-12-19T16:57:43Z" title="Thursday, December 19th 2019, 5:57:43 pm +01:00">Dec 19, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,69 @@
+const transformReturnType = record => {
+  return {
+    __typename: record.get('type'),
+    ...record.get('resource').properties,
+  }
+}
+export default {
+  Query: {
+    findResources: async (_parent, args, context, _resolveInfo) => {
+      const { query, limit } = args
+      const { id: thisUserId } = context.user
+      // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
+      const myQuery = query + '*'
```


I am not sure about this. It is not meant to search full phrases. The fulltext search is looking for each word everywhere and not for phrases. If you enter more then one word, then only the last word is searched with the wildcard. The other words must be written completely to be found. I am not sure if this is the purpose of a search.
I would prefer this one:
`const myQuery = query.replace(/\s+/g, '* ') + '*'`

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-20T11:32:45Z" title="Friday, December 20th 2019, 12:32:45 pm +01:00">Dec 20, 2019</time>_

---


we can add it back @Mogge , but it was kinda broken for me... so if I search for `not blocked` in the cypress test it would not find any posts with the content `blocked`, only if I search for `blocked` would it work.
I think maybe the only reason it started working as I expected is that with adding just the wildcard at the end of the phrase `not blocked` means that it is only looking for `blocked`, but at least it actually finds results.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-06T11:58:01Z" title="Monday, January 6th 2020, 12:58:01 pm +01:00">Jan 6, 2020</time>_

---


I have tested this out with `ökologischer Garten` in the Post content and searching for `öko garten` as you suggested in Discord @Mogge... I can see what you are concerned about.
I have added this line of code back :point_up_2:,  but now it is failing the cypress tests when searching for `not blocked` with the content being `cause I'm not blocked` ... what would you suggest?